### PR TITLE
Implement Hybrid AMM-CDA Router

### PR DESCRIPTION
**Pull request by Chralt98**

### CODE REVIEW

File: zrml/orderbook/src/lib.rs [hunk 1]
```diff
@@ -463,7 +464,10 @@ mod pallet {
         ) -> DispatchResult {
             let market = T::MarketCommons::market(&market_id)?;
             ensure!(market.status == MarketStatus::Active, Error::<T>::MarketIsNotActive);
-            ensure!(market.scoring_rule == ScoringRule::Orderbook, Error::<T>::InvalidScoringRule);
```
Comment (Robiquet) [inline zrml/orderbook/src/lib.rs:466]: Wouldn't `ScoringRule::Orderbook` still be valid?

Comment (Robiquet) [reply zrml/orderbook/src/lib.rs:466]: If every market is going to be initialised with `ScoringRule::AmmCdaHybrid` what will happen to parimutuel markets?

Comment (maltekliemann) [reply zrml/orderbook/src/lib.rs:466]: From https://hackmd.io/NBVKUY1QT--YnotR5KUx2w#Scoring-Rules: "The Lmsr and Orderbook variants are removed from the ScoringRule enum and replaced by AmmCdaHybrid, which allows a market to use both the order book and the AMM."

> If every market is going to be initialised with ScoringRule::AmmCdaHybrid what will happen to parimutuel markets?

The `ScoringRule::Parimutuel` object will not be deleted, so the parimutuel markets stay as they are.

**Comment by Chralt98** [comment]
Abandoned in favor of multiple smaller PRs!

**Pull request closed without merging**